XamlControlsGallery/ControlPages/AnimatedVisualPlayerPage.xaml.cs

```diff
@@ -48,18 +31,56 @@ private void ReverseButton_Click(object sender, RoutedEventArgs e)
 
         private void EnsurePlaying()
         {
-            if (PauseButton.IsChecked.Value)
+            if (!Player.IsPlaying)
             {
-                // Resume playing the animation, if paused.
-                PauseButton.IsChecked = false;
+                // Play the animation at the currently specified playback rate.
+                _ = Player.PlayAsync(fromProgress: 0, toProgress: 1, looped: false);
+                SetIsPlayingIndicator(true);
+            }
+        }
+
+        private void PlayPauseButton_Click(object sender, RoutedEventArgs e)
+        {
+            // Is the player playing and we did not get paused?
+            if(Player.IsPlaying && !wasPaused)
+            {
+                Player.Pause();
+                SetIsPlayingIndicator(false);
+                wasPaused = true;
             }
             else
             {
-                if (!Player.IsPlaying)
+                // Either not playing or paused
+                // If paused, resume and set paused to false
+                if(wasPaused)
                 {
-                    // Play the animation at the currently specified playback rate.
-                    _ = Player.PlayAsync(fromProgress: 0, toProgress: 1, looped: false);
+                    Player.Resume();
+                    wasPaused = false;
                 }
+                // Not playing, start animation now
+                else
+                {
+                    Player.PlaybackRate = 1;
+                    _ = Player.PlayAsync(fromProgress:0, toProgress: 1, looped: false);
```

Comment:
Why have the `_ =`? I know you didn't introduce this, but I'm wondering if it's necessary here and above.

Reply:
This is probably a stupid excuse, but it's the easiest way to let the C# compiler not raise the warning that "PlayAsync" is async and it might not have finished execution before the next line get's executed. By assigning the "result" of "PlayAsync" to "_" we essentially say that we care enough about the async task that we store it, but not enough to actually store it but rather discard the value. So at the end the C# compiler thinks something that makes him not complain about a non awaited async task.

Another way to fix this would be to add await or await the function another way.